validator/db/kv/attestation_history_new.go

```diff
@@ -150,3 +151,56 @@ func (store *Store) SaveAttestationHistoryNewForPubKeys(ctx context.Context, his
 	})
 	return err
 }
+
+// ImportOldAttestationFormat import old attestation format data into the new attestation format
+func (store *Store) ImportOldAttestationFormat(ctx context.Context) error {
+	ctx, span := trace.StartSpan(ctx, "Validator.ImportOldAttestationFormat")
+	defer span.End()
+	var allKeys [][]byte
+
+	if err := store.db.View(func(tx *bolt.Tx) error {
+		attestationsBucket := tx.Bucket(historicAttestationsBucket)
+		if err := attestationsBucket.ForEach(func(pubKey, _ []byte) error {
+			pubKeyCopy := make([]byte, len(pubKey))
+			copy(pubKeyCopy, pubKey)
+			allKeys = append(allKeys, pubKeyCopy)
+			return nil
+		}); err != nil {
+			return errors.Wrapf(err, "could not retrieve attestations for source in %s", store.databasePath)
+		}
+
+		return nil
+	}); err != nil {
+		return err
+	}
+	allKeys = removeDuplicateKeys(allKeys)
+	var keys [][48]byte
```

Comment:
if you need a variable sized array of [48] bytes, you should just do it above. You again copy it there and copy the pubkey again here.

Reply:
`removeDuplicateKeys` uses [][]byte

Reply:
not sure why that is the type used by a method called `removeDuplicateKeys` . All BLS pubkeys are 48 bytes, there is no variation in their size. If it is an easy change, we should do it if not you can just leave it as it is.

Reply:
fixed